### PR TITLE
Split compression plugin into zlib and zopfli plugins to avoid optional dependency for zlib users

### PR DESCRIPTION
**Pull request by palmerj3**

### CODE REVIEW

**Comment by palmerj3** [comment]
@bebraw @sokra any thoughts?

**Comment by bebraw** [comment]
It's up to @sokra to make the final call.

**Comment by palmerj3** [comment]
Not really interested in chasing people to look at PRs

File: README.md [hunk 1]
```diff
@@ -25,6 +31,27 @@ Arguments:
...
 * `minRatio`: Only assets that compress better that this ratio are processed. Defaults to `0.8`.
 
+
+### zopfli
+``` bash
+npm install --save-dev compression-webpack-plugin-zopfli
+```
+
+``` javascript
+var CompressionPlugin = require("compression-webpack-plugin");
```
Comment (SpaceK33z) [inline README.md]: Shouldn't this require be `compression-webpack-plugin-zopfli`?

File: README.md [hunk 2]
```diff
@@ -2,6 +2,12 @@
...
 ## Usage
 
+### zlib
+
+``` bash
+npm install --save-dev compression-webpack-plugin-zlib
+```
+
 ``` javascript
 var CompressionPlugin = require("compression-webpack-plugin");
```
Comment (SpaceK33z) [inline README.md]: Shouldn't this be `compression-webpack-plugin-zlib`?

**Comment by bebraw** [comment]
@palmerj3 We've split the loaders/plugins from the core org and are getting PRs in. Please let me know if you want to revive this PR.

**Comment by michael-ciniawsky** [comment]
@bebraw Yep I haven't started looking what options are available, but would it be possible to just do

**webpack.config.js**
```js
import Zopfli from 'zopfli'
import Brotli from 'brotli'

import CompressionPlugin from 'compression-webpack-plugin'

const config = {
   plugins: [
     new CompressionPlugin({ 
       algorithm: (buffer, options, cb) => Zopfli() // this should already be possible
     })
   ]
}
```

So `compression-webpack-plugin` ships with Node's zlib by default and other algos could be installed and registered as needed + documentation/how-to for Zopfli, Brotli etc

**Comment by bebraw** [comment]
@michael-ciniawsky Yeah, having a way to inject algorithms from outside is probably enough to satisfy this PR.

@palmerj3 Feel free to comment on this so we get it right.

**Comment by michael-ciniawsky** [comment]
@bebraw If there are good implementations of Zopfli and Brotli they can also be directly integrated but going with zlib is the savest imho

@palmerj3 Please help maintaining this repo, there will be a good balance of activity/communication from now on  👍 😛

**Pull request closed without merging**